scripts/22-starting-rust.md

```diff
@@ -0,0 +1,582 @@
+
+---
+
+## Rust by Example
+
+1. Hello World
+2. Primitives
+3. Custom Types
+4. Variable Bindings
+5. Types
+6. Conversion
+7. Expressions
+8. Flow of Control
+9. Functions
+10. Modules
+
+notes:
+
+Rust By Example is less linearly linked to The Book.
+Though it's partially in the same order, I read it as a supplement to The Book.
+It includes high-level tips on idiomatic error handling, how to organise your code into modules, and testing strategies.
+These are all covered in The Book, but Rust By Example is more opinionated on HOW to use them.
+
+Keep all three of these tools together.
+
+We'll get back to the main quest in a moment, but first, a quiz, from today's sponsor, Razor Secure:
+
+
+---
+
+<!-- slide bg="[[rs-train.jpg]]" -->
+
+![[rs-logo.png|400]]
+notes:
+_(disclosure: The company's CTO is my brother!)_
+
+What data-centre:
+- travels at 100 miles-per-hour
+- re-configures every IP when it attaches or detaches to other data-centres
+- all without a reliable power source or internet connection, and must never fail?
+
+The Answer: A train.
+
+---
+
+<!-- slide bg="[[rs-train.jpg]]" -->
+
+![[rs-logo.png|200]]
+
+1. Cross-platform Rust Agent
+2. Cloud microservices
+3. Embedded hardware platform
+
+notes:
+
+- RazorSecure is a 50-person startup bringing cutting-edge security tech to the rapidly-advancing world of rail
+- They do this through:
+   - A Rust intrusion detection and monitoring agent running on-board.
+   - A cloud environment running K8s, Python micro-services, and event-based data processing, and
+   - A Yocto hardware platform running custom embedded linux.
+- Their team and customers span Europe and North America, so if you have taken a train journey here, then RazorSecure's security systems may have already kept you safe.
+
+---
+
+<!-- slide bg="[[rs-train2.jpg]]" -->
+<split even>
+
+&nbsp;
+&nbsp;
+![[Python_logo_icon.png|200]]
+&nbsp;
+&nbsp;
+![[kubernetes-logo.png|200]]
+
+&nbsp;
+&nbsp;
+![[rust-logo.png|200]]
+
+</split>
+
+notes:
+
+- If you are excited by this challenge and tech, then they are VERY interested in speaking to you as they are hiring NOW.
+- The company is fully remote, so wherever you are based they offer challenging work in an interesting field with some awesome technology in a highly collaborative team with flexible working practises.
+
+---
+
+![[rs-logo.png|200]]
+
+[RazorSecure.com/careers](https://www.razorsecure.com/careers)
+
+notes:
+
+Find out more about jobs at RazorSecure at RazorSecure.com/careers.
+
+My thanks to RazorSecure for their support of this channel.
+
+Back to the side quest:
+
+---
+
+### SIDE QUEST #1
+
+![[ultralearning.png|300]]
+
+notes:
+
+This is Ultralearning by Scott H Young.
+(not sponsored)
+A very good book on the meta process of learning to learn.
+
+You should read it, or listen to it on audiobook as I did.
+
+---
+
+![[top-men-warehouse-indiana.png]]
+
+(http://science.sciencemag.org/content/331/6818/772)
```

Comment:
Sadly, this link http://science.sciencemag.org/content/331/6818/772 is `404` 💔 
Redirects to: https://www.science.org/doi/10.1126/science.72
<img width="1298" alt="image" src="https://user-images.githubusercontent.com/194400/227715516-43108031-6658-4202-aa8a-b0b43be4a51f.png">

Attempted to recover it via https://archive.org/ but no luck. 😞